### ISSUE TITLE
float type support ".xxx" syntax

### ISSUE THREAD

**Issue by winkyao** [pingcap/tidb#3889]
```
create table float_test(id float not null, string varchar(64) not null);
insert float_test values(.67, 'string');
```

MySQL:

```
mysql> create table float_test(id float not null, string varchar(64) not null);
Query OK, 0 rows affected (0.01 sec)

mysql> insert float_test values(.67, 'string');
Query OK, 1 row affected (0.00 sec)
```

TiDB:

```
mysql> create table float_test(id float not null, string varchar(64) not null);
Query OK, 0 rows affected (0.09 sec)

mysql> insert float_test values(.67, 'string');
ERROR 1105 (HY000): line 0 column 26 near "67, 'string')" (total length 39)
```

**Comment by breeswish** [comment]
Generalized to issue 3954.

**Comment by breeswish** [comment]
Fixed by #3964
